### PR TITLE
handle_out() returns even when out is set

### LINKED ISSUE #8285: handle_out() returns nothing if out is set
**What happened**: function set o to the output dask array and returned None.

**What you expected to happen**: function set o to the output dask array and return the output array.

**Minimal Complete Verifiable Example**:

```py
>>> from dask import array as da
>>> x = da.array( [1,2,3] )
>>>
>>> print( da.invert( x ) )
dask.array<invert, shape=(3,), dtype=int64, chunksize=(3,), chunktype=numpy.ndarray>
>>>
>>> o = da.array( [4,5,6] )
>>> print( da.invert( x, out=o ) )
None
```

**Anything else we need to know?**:

**Environment**:

- Dask version: dask-2021.9.1+38.geab6a998
- Python version: 3.9.7
- Operating System: Manjaro
- Install method (conda, pip, source): source

### PR DESCRIPTION
**Pull request by ek234**
Closes #8285

### CODE REVIEW

File: dask/array/core.py [hunk 1]
```diff
@@ -4480,8 +4480,7 @@ def handle_out(out, result):
             " Received type %s, expected Dask Array" % type(out).__name__
         )
         raise NotImplementedError(msg)
-    else:
-        return result
+    return result
```
Comment (douglasdavis) [inline dask/array/core.py:4483]: While this should eventually yields the correct result after a Dask computation, I think it follows NumPy behavior to return the actual `out` object under the `if isinstance(out, Array):` branch, that way the returned object is the same as the object passed to the `out=` argument.

**Review by douglasdavis — COMMENTED** [review]
Thanks @ek234! I left one comment about your code fix, another thing we should include is a new test to validate the new behavior. [dask/array/tests/test_array_core.py](https://github.com/dask/dask/blob/main/dask/array/tests/test_array_core.py) would be an appropriate place to add a new test to make sure `out=` returns the same object passed in as the argument.

**Comment by ek234** [comment]
@douglasdavis  is this what you were thinking?

**Comment by douglasdavis** [comment]
Indeed, and let's revert the first change (removing the `else` block) so `git blame` is limited to the adjustment being made to the `out` case-- with that and a test this should be good to go.

File: dask/array/tests/test_array_core.py [hunk 2]
```diff
@@ -918,6 +918,29 @@ def test_elemwise_dtype():
...
+            [11, 2, 3, 4, 5, 12, 34, 56, 78, 90],
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+        ],
+        [
+            [[11, 2, 3, 4, 5], [6, 7, 8, 9, 0], [12, 34, 56, 78, 90]],
+            [[1, 1, 1, 1, 1], [1, 1, 1, 1, 1], [1, 1, 1, 1, 1]],
+        ],
+    ],
+)
+def test_elemwise_return(x, o):
```
Comment (douglasdavis) [inline dask/array/tests/test_array_core.py:934]: A better name might be something like `test_handle_out_return`; `elemwise` isn't the only user of `handle_out`

File: dask/array/tests/test_array_core.py [hunk 3]
```diff
@@ -918,6 +918,29 @@ def test_elemwise_dtype():
...
+
+    x = da.array(x, dtype=np.int8)
+    o = da.array(o, dtype=np.int8)
+
+    o_ = da.invert(x, out=o)
+
```
Comment (douglasdavis) [inline dask/array/tests/test_array_core.py:940]: an additional `assert o_ is o` line would test that the object returned is the same as the object passed to `out=`

**Review by douglasdavis — COMMENTED** [review]
Added a couple more comments, looks like the PR is almost good to go!

File: dask/array/tests/test_array_core.py (hunk 2 above)
Comment (ek234) [reply dask/array/tests/test_array_core.py]: ohh i searched in the file and saw it is used only once. Now, I saw it is used in different files.

**Comment by ian-r-rose** [comment]
@ek234 Is this superseded by #8253? If so, can this PR be safely closed?

**Comment by jcrist** [comment]
Yes, this is superseded by #8253. Closing.

**Pull request closed without merging**